### PR TITLE
[PLT-6652] tests for MenuList component

### PR DESCRIPTION
**Pull request by licarm**
## Description
tests for MenuList component

## Notes
- keydown events simulated by enzyme won't work with `window.addEventListener()` because enzyme tests React elements, so enzyme's `simulate('keydown')` will not be detected by the listener.

### CODE REVIEW

File: packages/riipen-ui/src/components/MenuList.test.jsx [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+      wrapper
+        .find("MenuList")
+        .childAt(0)
+        .simulate("click");
+
+      expect(selectChange).toHaveBeenLastCalledWith(null, expect.any(Object));
+    });
+
+    describe("when variant is selection", () => {
```
Comment (imogenem) [inline packages/riipen-ui/src/components/MenuList.test.jsx:182]: It looks like this describe block could be removed and the test could be moved to variant. The test name for this test and the test at line 262 sound very similar.

File: packages/riipen-ui/src/components/MenuList.test.jsx [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+      const selectChange = jest.fn();
+
+      mount(
+        <MenuList autoFocus={autoFocus} selectChange={selectChange}>
+          <MenuItem>one</MenuItem>
+          <MenuItem>two</MenuItem>
+        </MenuList>
+      );
+
+      expect(selectChange).toHaveBeenLastCalledWith(0, undefined);
```
Comment (imogenem) [inline packages/riipen-ui/src/components/MenuList.test.jsx:65]: Would this test pass with just one argument, the zero? 

`toHaveBeenLastCalledWith(0)`

**Review by imogenem — CHANGES_REQUESTED** [review]
This is a tough file to test, so great job. I'm wondering what the code coverage will look like for this file, and if more things needs to be/can be tested, like keyDownHandler().

File: packages/riipen-ui/src/components/MenuList.jsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ class MenuList extends React.Component {
 
   static propTypes = {
     /**
-     * Wether to select the first element on list open
+     * Whether to select the first element on list open
```
[the remark this answers is not in the document]
Comment (jordanell) [reply packages/riipen-ui/src/components/MenuList.jsx:13]: 10 points for Gryffindor

File: packages/riipen-ui/src/components/MenuList.test.jsx [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+      const component = wrapper.find("MenuList");
+      expect(component.props().selectedIndex).toEqual(
+        defaultProps.selectedIndex
+      );
+      expect(component.props().variant).toEqual(defaultProps.variant);
+    });
+  });
+
+  describe("autoFocus prop", () => {
+    it("selects first element on list open with true autoFocus and default selectedIndex", () => {
```
Comment (jordanell) [inline packages/riipen-ui/src/components/MenuList.test.jsx:54]: This sentence is sort of confusing to me. I think what we are testing is that we auto focus the first element of the list on mount, so what if we said "selects first item of list with ..."

File: packages/riipen-ui/src/components/MenuList.test.jsx [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+          .at(0)
+          .childAt(0)
+          .simulate("click");
+
+        expect(selectChange).not.toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe("selectedIndex prop", () => {
```
Comment (jordanell) [inline packages/riipen-ui/src/components/MenuList.test.jsx:223]: Should we add a test for selectedIndex + autoFocus? I don't see this combination.

File: packages/riipen-ui/src/components/MenuList.test.jsx (hunk 1 above)
Comment (jordanell) [reply packages/riipen-ui/src/components/MenuList.test.jsx:182]: Ya I am also confused about this describe block. Looking at the two tests within it, I wonder if they are better off being part of the children tests, or even remove the describe block entirely and just put those two tests at the same level of the other selectChange test. Up to you.

**Review by jordanell — CHANGES_REQUESTED** [review]
Nice. I think we are missing some tests for the `keyDownHandler` code as part of the component though.

File: packages/riipen-ui/src/components/MenuList.test.jsx (hunk 1 above)
Comment (licarm) [reply packages/riipen-ui/src/components/MenuList.test.jsx:182]: I like them being part of children tests

File: packages/riipen-ui/src/components/MenuList.test.jsx (hunk 2 above)
Comment (licarm) [reply packages/riipen-ui/src/components/MenuList.test.jsx:65]: no :(

File: packages/riipen-ui/src/components/MenuList.test.jsx [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+
+      expect(
+        wrapper
+          .find("MenuItem")
+          .at(0)
+          .props().selected
+      ).toBeUndefined();
+    });
+
+    it("prioritizes selectedIndex over autoFocus when autoFocus is true", () => {
```
Comment (jordanell) [inline packages/riipen-ui/src/components/MenuList.test.jsx:258]: Nice

File: packages/riipen-ui/src/components/MenuList.test.jsx [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+          <MenuItem>one</MenuItem>
+          <MenuItem>two</MenuItem>
+        </MenuList>
+      );
+
+      expect(selectChange).not.toHaveBeenCalled();
+      expect(
+        wrapper
+          .find("MenuItem")
+          .at(selectedIndex)
```
Comment (jordanell) [inline packages/riipen-ui/src/components/MenuList.test.jsx:279]: Slick

**Review by jordanell — APPROVED** [review]

**Review by imogenem — APPROVED** [review]

**Pull request merged**